pkg/networkutils/network.go

```diff
@@ -72,32 +71,24 @@ func isDuplicateRuleAdd(err error) bool {
 // TODO : implement ip rule not to 10.0.0.0/16(vpc'subnet) table main priority  1024
 func (os *linuxNetwork) SetupHostNetwork(vpcCIDR *net.IPNet, primaryAddr *net.IP) error {
 
-	//TODO : temporary implement "ip rule not to vpc cider table main priority 1024
-	cmd := append([]string{"ip", "rule", "add", "not", "to", vpcCIDR.String(), "table", "main", "priority", "1024"})
+	hostRule := os.netLink.NewRule()
+	hostRule.Dst = vpcCIDR
+	hostRule.Table = mainRoutingTable
+	hostRule.Priority = hostRulePriority
+	hostRule.Invert = true
+
+	// if this is a restart, cleanup previous rule first
+	err := os.netLink.RuleDel(hostRule)
```

Comment:
Why not just invoke RuleAdd(...) and check if error code contains anything other than ENOENT?

Reply:
hmmm, in some OS (e.g. ubuntu) it will create 2 Rule entries and not complaining anything.